lib/Factory/MenuFactory.php

```diff
@@ -4,88 +4,142 @@
 
 use Timber\CoreInterface;
 use Timber\Menu;
-
 use WP_Term;
 
 /**
  * Internal API class for instantiating Menus
  */
 class MenuFactory {
-	public function from($params, array $args = []) {
-		if ($params === 0) {
-			// We know no nav_menu term exists with ID 0,
-			// so just look at the existing terms in the database.
+
+	/**
+	 * Try to get a menu by all means available in an order that matches the
+	 * most common use cases. It will fallback on the first menu found if no parameters are provided
```

Comment:
The idea behind this was to mimic the `wp_nav_menu` behaviour which tries to find a menu by all means available.

Honestly, I'll personnaly never use that method so I'm fine with either option, just thought it would bring more parity with `wp_nav_menu`.

Reply:
Yeah, that’s the way to usually do it. But because this has led to unexpected behaviors and problems, I’d like to follow what we discussed https://github.com/timber/timber/issues/2089#issuecomment-586739839 and implement the changes listed in #2292.